### PR TITLE
Take result attributes directly from the production when resolving contexts

### LINKED ISSUE #4683: [K-Bug] Broken condition for heating rules with result
### What component is the issue in?

Front-End

### Which command

- [X] kompile
- [ ] kast
- [ ] krun
- [ ] kprove
- [ ] kprovex
- [ ] ksearch

### What K Version?

v7.1.164-0-g459fdd7b84

### Operating System

Linux

### K Definitions (If Possible)

```k
module A
  imports BOOL

  syntax KResult

  syntax Stuff ::= "a" | "b" | "c"
  syntax KItem ::= thing(Stuff, Stuff)  [seqstrict, result(MyResult)]

  rule thing(c, c) => .K

  rule b => c
  rule a => c

  syntax Bool ::= isMyResult(K)  [function, total, symbol(isMyResult)]
  rule isMyResult(_) => false  [owise]
  rule isMyResult(c) => true


endmodule
```

### Steps to Reproduce

```
echo "thing(a, b)" > a.in
kompile a.k
krun a.in
```

Note that execution stops without heating the second argument of "thing". As far as I can tell, this happens because the second heating rule (the one for "b") requires that the first argument is KResult instead of MyResult. 

### Expected Results

The second argument of "thing" should be heated, then evaluated to "c" and put back, then "thing(c, c)" should be rewritten to ".K". That is, the end result should be:
```
<k>
  .K
</k>

```

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
A declaration like
```syntax KItem ::= thing(Stuff, Stuff)  [seqstrict, result(Evaluated)]```
should consider tems that are `Evaluated` as evaluated. Right now, this happens only for the argument being heated/cooled, while the previous arguments are checked for being `KResult`. As an example, when heating the second argument above, the heating rule looks something like this:
```
rule thing(A, B) => B ~> thingFreezer(A) requires isKResult(A) andBool notBool isEvaluated(B)
```

With this PR, the rule becomes
```
rule thing(A, B) => B ~> thingFreezer(A) requires isEvaluated(A) andBool notBool isEvaluated(B)
```

Fixes #4683

### CODE REVIEW

File: k-frontend/src/main/java/org/kframework/compile/ResolveStrict.java [hunk 1]
```diff
@@ -119,6 +119,14 @@ private void setAliases(String attribute, Set<ContextAlias> aliases, Production
   private static final ContextAlias DEFAULT_ALIAS =
       ContextAlias(KVariable("HERE"), BooleanUtils.TRUE, Att.empty());
 
+  private static final ContextAlias defaultAliasWithAtts(Att att) {
+    if (!att.contains(Att.RESULT())) {
+      return DEFAULT_ALIAS;
```
Comment (tothtamas28) [inline k-frontend/src/main/java/org/kframework/compile/ResolveStrict.java]: If `DEFAULT_ALIAS` is not referred to elsewhere, consider inlining it here.

File: k-frontend/src/main/java/org/kframework/compile/ResolveStrict.java [hunk 2]
```diff
@@ -119,6 +119,14 @@ private void setAliases(String attribute, Set<ContextAlias> aliases, Production
   private static final ContextAlias DEFAULT_ALIAS =
       ContextAlias(KVariable("HERE"), BooleanUtils.TRUE, Att.empty());
 
+  private static final ContextAlias defaultAliasWithAtts(Att att) {
```
Comment (tothtamas28) [inline k-frontend/src/main/java/org/kframework/compile/ResolveStrict.java]: Consider

```suggestion
  private static final ContextAlias defaultAliasFor(Production production) {
```

File: k-distribution/tests/regression-new/issue-4683/test.k [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  rule thing(c, c) => .K
+
+  rule b => c
+  rule a => c
+
+  syntax Bool ::= isMyResult(K)  [function, total, symbol(isMyResult)]
+  rule isMyResult(_) => false  [owise]
+  rule isMyResult(c) => true
+
+
```
Comment (tothtamas28) [inline k-distribution/tests/regression-new/issue-4683/test.k]: Nitpick: consider removing these blank lines.

**Review by tothtamas28 — APPROVED** [review]
Before merging, please add a description for the PR, as that will become the commit description once it is merged into `master`.

File: k-frontend/src/main/java/org/kframework/compile/ResolveStrict.java (hunk 1 above)
Comment (virgil-serbanuta) [reply k-frontend/src/main/java/org/kframework/compile/ResolveStrict.java]: Done

File: k-distribution/tests/regression-new/issue-4683/test.k (hunk 3 above)
Comment (virgil-serbanuta) [reply k-distribution/tests/regression-new/issue-4683/test.k]: Done

**Comment by virgil-serbanuta** [comment]
> Before merging, please add a description for the PR, as that will become the commit description once it is merged into `master`.

I hope it looks reasonable now.

**Pull request merged**